ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -164,307 +219,272 @@ private HelixBootstrapUpgradeUtil(String hardwareLayoutPath, String partitionLay
+          // availability and so on should be updated at all (if not, instanceConfigFromStatic should be replaced with
+          // the appropriate instanceConfig that is constructed with the correct values from both.
+          if (!dryRun) {
+            dcAdmin.setInstanceConfig(clusterName, instanceName, instanceConfigFromStatic);
+          }
+        } else {
+          info("Instance {} already present in Helix, with same InstanceConfig, skipping. Remaining instances: {}",
+              instanceName, --totalInstances);
         }
       }
-    }
-    if (!partitionsUnderNextResource.isEmpty()) {
-      addNewAmbryPartitions(partitionsUnderNextResource, Long.toString(nextResource));
-    }
-  }
 
-  /**
-   * Initialize a map of dataCenter to HelixAdmin based on the given zk Connect Strings.
-   * @param helixAdminFactory the {@link HelixAdminFactory} to use to instantiate {@link HelixAdmin}
-   */
-  private void initializeAdminsAndAddCluster(HelixAdminFactory helixAdminFactory) {
-    for (Map.Entry<String, ClusterMapUtils.DcZkInfo> entry : dataCenterToZkAddress.entrySet()) {
-      HelixAdmin admin = helixAdminFactory.getHelixAdmin(entry.getValue().getZkConnectStr());
-      adminForDc.put(entry.getKey(), admin);
-      // Add a cluster entry in every DC
-      if (!admin.getClusters().contains(clusterName)) {
-        admin.addCluster(clusterName);
-        admin.addStateModelDef(clusterName, LeaderStandbySMD.name, LeaderStandbySMD.build());
+      for (String instanceName : instancesInStatic) {
+        InstanceConfig instanceConfigFromStatic =
+            createInstanceConfigFromStaticInfo(dcToInstanceNameToDataNodeId.get(dc.getName()).get(instanceName),
+                partitionsToInstancesInDc);
+        info("Instance {} is new, adding to Helix. Remaining instances: {}", instanceName, --totalInstances);
+        if (!dryRun) {
+          dcAdmin.addInstance(clusterName, instanceConfigFromStatic);
+        }
       }
-    }
-  }
 
-  /**
-   * Populate the set of existing resources and existing partitions in the cluster. This assumes that all partitions
-   * and resources exist in all datacenters (This assumption helps simplify the logic. This can be gotten rid of in
-   * the future if need be).
-   * @param dcAdmin the reference admin (preferably the admin to the zookeeper server in the local datacenter).
-   */
-  private void populateResourcesAndPartitionsSet(HelixAdmin dcAdmin) {
-    for (String resource : dcAdmin.getResourcesInCluster(clusterName)) {
-      existingResources.add(Long.valueOf(resource));
-      for (String partition : dcAdmin.getResourceIdealState(clusterName, resource).getPartitionSet()) {
-        existingPartitions.add(Long.valueOf(partition));
+      for (String instanceName : instancesInHelix) {
+        info("Instance {} is in Helix, but not in static. Ignoring for now. Remaining instances: {}", --totalInstances);
```

Comment:
Here miss a `instanceName` in the info. 
`info("Instance {} is in Helix, but not in static. Ignoring for now. Remaining instances: {}", instanceName, --totalInstances);`

Reply:
Fixed, thanks.